compiler/rustc_type_ir/src/search_graph/mod.rs

```diff
@@ -0,0 +1,597 @@
+use crate::data_structures::{HashMap, HashSet};
+use crate::solve::SolverMode;
+use rustc_index::{Idx, IndexVec};
+use std::fmt::Debug;
+use std::hash::Hash;
+use std::marker::PhantomData;
+use std::mem;
+use tracing::debug;
+
+mod global_cache;
+use global_cache::CacheData;
+pub use global_cache::GlobalCache;
+mod validate;
+
+pub trait Cx: Copy {
```

Comment:
Would be nice to leave a comment here explaining why we have yet another context

Reply:
Specifically, why it's better to duplicate its methods and having a blanket impl or having this be a supertrait or something.